### PR TITLE
Reworked Platform Props Exporting

### PR DESCRIPTION
**Pull request by andreiborodin**
This follows the same approach as #109 in updating how the exporter logic is split up for a cleaner and well understood split of responsibilities.

### CODE REVIEW

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildUnityProjectExporter.cs [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        /// <summary>
+        /// Exports the core Unity props file.
+        /// </summary>
+        /// <param name="exporter">The overal exporter to use for creating exporters.</param>
+        /// <param name="platform">The compilation platform to export.</param>
+        /// <param name="inEditor">Whether to export the inEditor version.</param>
+        public static void ExportCoreUnityPropFile(IUnityProjectExporter exporter, CompilationPlatformInfo platform, bool inEditor)
+        {
+            string configuration = inEditor ? "InEditor" : "Player";
```
Comment (chrisfromwork) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildUnityProjectExporter.cs]: >"InEditor" : "Player" [](start = 46, length = 21)

it feels like these should be hard coded as constants somewhere for consistency throughout the exporters

**Comment by chrisfromwork** [comment]
> // Copyright (c) Microsoft Corporation. All rights reserved.

nit: remove "All rights reserved."

---
Refers to: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IPlatformPropsExporter.cs:1 in 3252586. [](commit_id = 3252586a49728740f2b8fae725c9acefdd0ba203, deletion_comment = False)

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IUnityProjectExporter.cs [hunk 2]
```diff
@@ -46,11 +46,20 @@ public interface IUnityProjectExporter
...
+        /// <param name="unityPlatform">The unity platform for the platform props.</param>
+        /// <param name="scriptingBackend">The scripting backend for the platform props.</param>
+        IPlatformPropsExporter CreatePlatformPropsExporter(FileInfo path, string unityConfiguration, string unityPlatform, ScriptingBackend scriptingBackend);
+
+        /// <summary>
+        /// Creates the specialized platform props file exporter for Player|WSA combination.
+        /// </summary>
+        /// <param name="path">The <see cref="FileInfo"/> representing where this props file will be written.</param>
+        /// <param name="scriptingBackend">The scripting backend for the platform props.</param>
+        IWSAPlayerPlatformPropsExporter CreateWSAPlayerPlatformPropsExporter(FileInfo path, ScriptingBackend scriptingBackend);
```
Comment (chrisfromwork) [inline Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IUnityProjectExporter.cs]: >CreateWSAPlayerPlatformPropsExporter [](start = 40, length = 36)

are we going to need a custom exporter for each platform/player combination? Are there specific things needed on IWSAPlayerPlatformPropsExporter that wouldn't pertain to other platforms?

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildUnityProjectExporter.cs (hunk 1 above)
Comment (andreiborodin) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/MSBuildUnityProjectExporter.cs]: Possibly, but will save that for a later change.

File: Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IUnityProjectExporter.cs (hunk 2 above)
Comment (andreiborodin) [reply Source/MSBuildTools.Unity/Packages/com.microsoft.msbuildforunity/Editor/ProjectGenerator/Scripts/Exporters/IUnityProjectExporter.cs]: Yes, and yes Min/Max target UWP platform.

**Comment by andreiborodin** [comment]
Putting it in, will remove the copyright in a subsequent PR I will be putting out soon.

**Pull request merged**